lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -380,7 +380,9 @@ func (handler *ContainersHandlersImpl) ContainerWaitHandler(params containers.Co
 
 	select {
 	case <-c.WaitForState(exec.StateStopped):
+		c.Refresh(context.Background())
```

Comment:
We already have a Refresh on this path, however it's got ordering issues:
see exec.go:145 and exec.go:156.

perhaps we should be moving the SetState call after the refresh (for success or failure)? The only one it's crucial for is stopped (exit code) but without knowing what the perf impact is I'm leaning towards state consistency.